dependency on libc must be explicitly specified in the build command when extern "ksuser" function is declared

### Zig Version

0.14.0-dev.2850+0ead0beb8

### Steps to Reproduce and Observed Behavior



`bug.zig`:
```zig
pub extern "ksuser" fn foo() u32;
comptime {
    _ = foo;
}
```
```sh
> zig test -target x86_64-windows bug.zig
bug.zig:1:12: error: dependency on libc must be explicitly specified in the build command
```

### Expected Behavior

The test should compile successfully, no linking to libc is required.

Works as expected on zig 0.13.0.

--- a/lib/std/zig/target.zig
+++ b/lib/std/zig/target.zig
@@ -195,11 +195,7 @@ pub fn isLibCLibName(target: std.Target, name: []const u8) bool {
             return true;
         if (eqlIgnoreCase(ignore_case, name, "ksguid"))
             return true;
-        if (eqlIgnoreCase(ignore_case, name, "ksuser"))
-            return true;
         if (eqlIgnoreCase(ignore_case, name, "largeint"))
-            return true;
-        if (eqlIgnoreCase(ignore_case, name, "locationapi"))
             return true;
         if (eqlIgnoreCase(ignore_case, name, "m"))
             return true;
@@ -213,13 +209,7 @@ pub fn isLibCLibName(target: std.Target, name: []const u8) bool {
             return true;
         if (eqlIgnoreCase(ignore_case, name, "moldname"))
             return true;
-        if (eqlIgnoreCase(ignore_case, name, "msxml2"))
-            return true;
-        if (eqlIgnoreCase(ignore_case, name, "msxml6"))
-            return true;
         if (eqlIgnoreCase(ignore_case, name, "msvcrt-os"))
-            return true;
-        if (eqlIgnoreCase(ignore_case, name, "ntoskrnl"))
             return true;
         if (eqlIgnoreCase(ignore_case, name, "portabledeviceguids"))
             return true;

